javascript/modules/client/src/context.ts

```diff
@@ -71,12 +71,8 @@ export class Context {
       gasFeeEstimationFactor: Context.resolveGasFeeEstimationFactor(
         contextParams.gasFeeEstimationFactor,
       ),
-      ipfs: contextParams.ipfsNodes.map((config) =>
-        new IpfsClient(config.url, config.headers)
-      ),
-      graphql: contextParams.graphqlNodes.map((url) =>
-        new GraphQLClient(url)
-      ),
+      ipfs: this.useIpfs(contextParams.ipfsNodes),
+      graphql: this.useGraphql(contextParams.graphqlNodes),
```

Comment:
```suggestion
      ipfs: contextParams.ipfsNodes,
      graphql: contextParams.graphqlNodes,
```

Reply:
Adapt according to the above

Reply:
Cannot do this, both graphql and ipfs typing is either a `GraphqlClient[]` or a `IpfsClient[]`, and the `contextParams.ipfsNodes` contains an object with the url and the headers. The `useIpfs` goes through this list creating the clients

Reply:
My point is that something like `useIpfs()` or `useGraphql()` sounds like a setter. But here is being used as a helper that transforms some input into some output. 

I would have:
- `useWeb3()`, `useIpfs()` and `useGraphql()` as the setters to update the state
- These would receive the context param types and internally use something like `digestWeb3Context()` or similar to do the "helper" part of transforming stuff

However, this may make more sense in the scope of the client than on the context one? WDYT?

Reply:
I believe it's a matter of naming, as you say, the `use...` naming maybe does not make much sense for these functions, maybe we can change it to something like `get...Clients` or something more clear since this functions transform `urls` into `clients`.

I then believe that a function for setting the `web3` or `ipfs` doer not make much sense, we already have that with the `set` function that receives a `Partial` of the `ContextParams` this would imply 2 ways of doing the exact same thing